tide_media.install

```diff
@@ -32,3 +36,104 @@ function tide_media_install() {
       ->save();
   }
 }
+
+/**
+ * SDPA-309: Associated current media items to Creative Commons Attribution 4.0.
+ */
+function tide_media_update_8002() {
+  $field_name = 'field_media_license';
+  $bundle_names = [
+    "audio",
+    "document",
+    "embedded_video",
+    "file",
+    "image",
+    "video",
+  ];
+  // First, lets find all media entities that still use old license field value.
+  $existingMedia = \Drupal::entityQuery('media')
+    ->exists($field_name)
+    ->execute();
+  if ($existingMedia && count($existingMedia)) {
+    // At this point we know there is media that uses old license field.
+    // Import the replacement field configuration.
+    module_load_include('inc', 'tide_core', 'includes/helpers');
+    $config_location = [drupal_get_path('module', 'tide_media') . '/config/install'];
+    $optional_location = [drupal_get_path('module', 'tide_media') . '/config/optional'];
+    // Import new vocabulary.
+    _tide_import_single_config('taxonomy.vocabulary.license_type', $config_location, TRUE);
+    // Import new field storage config.
+    _tide_import_single_config('field.storage.media.field_license_type', $config_location, TRUE);
+    // Import updated media browser view.
+    _tide_import_single_config('views.view.tide_media_browser', $config_location, TRUE);
+
+    foreach ($bundle_names as $bundle) {
+      _tide_import_single_config('field.field.media.' . $bundle . '.field_license_type', $config_location, TRUE);
+      _tide_import_single_config('core.entity_form_display.media.' . $bundle . '.default', $config_location, TRUE);
+      _tide_import_single_config('core.entity_view_display.media.' . $bundle . '.default', $config_location, TRUE);
+      _tide_import_single_config('jsonapi_extras.jsonapi_resource_config.media--' . $bundle, $optional_location, TRUE);
```

Comment:
@barbun this config requires JSON:API Extras module.

Reply:
Thanks @sonnykt . Addressed your comment.